src/hera/workflows/parameter.py

```diff
@@ -54,7 +54,7 @@ def as_input(self) -> _ModelParameter:
             value_from=self.value_from,
         )
 
-    def as_argument(self) -> Optional[_ModelParameter]:
+    def as_argument(self) -> _ModelParameter:
```

Comment:
Can we rename this to `as_parameter`? A bit confusing on the surface of it as we use it when creating a `parameter` under an `arguments`.

Reply:
Also as we're trying to remove the reuse of a single parameter in inputs/outputs/arguments

Reply:
Deciding against this in the end to match the `as_input`/`as_output`